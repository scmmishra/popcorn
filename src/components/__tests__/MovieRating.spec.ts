import { describe, it, expect } from "vitest";

import { mount, VueWrapper } from "@vue/test-utils";
import MovieRatingVue from "../MovieRating.vue";

function toFixedNumber(num: number, digits = 2) {
  let pow = Math.pow(10, digits);
  return Math.round(num * pow) / pow;
}

function getComponent(value?: number) {
  return mount(MovieRatingVue, {
    props: { value },
  });
}

describe("MovieRating", () => {
  it("passes brute force rendering", () => {
    // This is a brute force rendering test.
    // It's not very useful but validates for
    // sure that the component renders
    // the correct percentage value each time

    Array.from(Array(100))
      .map(() => toFixedNumber(Math.random() * 10))
      .forEach((value) => {
        const percentageValue = Math.round((value / 10) * 100);
        const wrapper = getComponent(value);

        expect(wrapper.html()).toContain(`data-pct="${percentageValue}"`);
      });
  });

  it.only("matches snapshot", () => {
    // This is a snapshot test to ensure that any
    // generated svg from the 0 to 10 range is valid
    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].forEach((value) => {
      let wrapper = getComponent(10);
      expect(wrapper.html()).toMatchSnapshot();

      wrapper = getComponent();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  it("returns errors for invalid values", () => {
    let wrapper = getComponent(4);
  });
});
