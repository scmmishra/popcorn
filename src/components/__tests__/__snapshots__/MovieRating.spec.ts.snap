// Vitest Snapshot v1

exports[`MovieRating > matches snapshot 1`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 2`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 3`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 4`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 5`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 6`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 7`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 8`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 9`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 10`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 11`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 12`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 13`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 14`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 15`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 16`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 17`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 18`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 19`] = `
"<div class=\\"circle-progress\\" data-pct=\\"100\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"0\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;

exports[`MovieRating > matches snapshot 20`] = `
"<div class=\\"circle-progress\\" data-pct=\\"0\\" data-v-ebb1aca2=\\"\\"><svg width=\\"36\\" height=\\"36\\" viewPort=\\"0 0 36 36\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-ebb1aca2=\\"\\">
    <circle stroke=\\"#164e63\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" data-v-ebb1aca2=\\"\\"></circle>
    <circle class=\\"progress_circle\\" stroke=\\"#22d3ee\\" r=\\"16\\" cx=\\"18\\" cy=\\"18\\" stroke-width=\\"4\\" fill=\\"none\\" stroke-dasharray=\\"100.53096491487338\\" stroke-dashoffset=\\"100.53096491487338\\" data-v-ebb1aca2=\\"\\"></circle>
  </svg></div>"
`;
